pylabrobot/liquid_handling/backends/hamilton/STAR_backend.py

```diff
@@ -5554,7 +5555,86 @@ async def core_move_plate_to_position(
 
     return command_output
 
-  # TODO:(command:ZB)
+  async def _star_core_read_barcode(
+    self,
+    slot_number: int,
+    minimal_z_position: int = 2200,
+    traverse_height_at_beginning_of_a_command: int = 2750,
+    z_speed: int = 1287,
+  ):
+    """Read a 1D barcode using the CoRe gripper scanner.
+
+    Returns:
+      A Barcode if one is successfully read, otherwise None.
+    """
+
+    assert 1 <= slot_number <= 54, "slot_number must be between 1 and 54"
+    assert 0 <= minimal_z_position <= 3600, "minimal_z_position must be between 0 and 3600"
+    assert (
+      0 <= traverse_height_at_beginning_of_a_command <= 3600
+    ), "traverse_height_at_beginning_of_a_command must be between 0 and 3600"
+    assert 0 <= z_speed <= 1287, "z_speed must be between 0 and 1287"
+
+    command_output = await self.send_command(
+      module="C0",
+      command="ZB",
+      cp=f"{slot_number:02}",
+      zb=f"{minimal_z_position:04}",
+      th=f"{traverse_height_at_beginning_of_a_command:04}",
+      zy=f"{z_speed:04}",
+      bd=1,
+      ma="0250 2100 0860 0200",
+      mr=0,
+      mo="000 000 000 000 000 000 000",
+    )
+
+    if command_output is None:
+      return None
+
+    resp = command_output.strip()
+    er_index = resp.find("er")
+    if er_index == -1:
+      # Unexpected format: no error section present.
+      raise ValueError(f"Unexpected CoRe barcode response (no error section): {resp}")
+
+    error_code = resp[er_index + 2 : er_index + 7]
+    if error_code != "00/00":
+      # Non-zero error code: raise a firmware error so the user sees it.
+      self.check_fw_string_error(resp)
+      return None
```

Comment:
```suggestion
    self.check_fw_string_error(resp)
```

will check errors more generally (sometimes firmware responses are odd and this method is made to handle all of them)